packages/ui/src/accounts/hooks/useIsVoteStakeLocked.ts

```diff
@@ -0,0 +1,12 @@
+import { useMemo } from 'react'
+
+import { Member } from '@/memberships/types'
+
+export const useIsVoteStakeLocked = (candidate?: Member, electionFinished?: boolean) => {
+  const isCouncilMember = candidate?.isCouncilMember
+  return useMemo(() => {
+    // Lock stake if the vote was cast: in current election or to winning candidate
+    // Enable stake recovery if election is finished
+    return !!candidate && (!electionFinished || isCouncilMember)
+  }, [electionFinished, candidate])
+}
```

Comment:
It's better now but it's still missing the case where a member which was candidate in a previous election (not the latest one) is now a council member. This is why I suggested `isLatestElection`. I'm thinking of something like this:

```suggestion
// Lock stake if the vote was cast: in current election or to winning candidate
// Enable stake recovery if election is finished
export const useIsVoteStakeLocked = (candidate?: Member, { isElectionFinished = true, isLatestElection = true }) =>
  isLatestElection && !!candidate && (!isElectionFinished || candidate.isCouncilMember)
```